src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -430,10 +430,14 @@ protected void writeToNBT(@NotNull final NBTTagCompound compound)
         compound.setTag(TAG_FREE_POSITIONS, freePositionsTagList);
 
         compound.setDouble(TAG_HAPPINESS, overallHappiness);
-        compound.setInteger(TAG_ABANDONED, lastContactInHours);
+        compound.setInteger(TAG_ABANDONED, packageManager.getLastContactInHours());
         compound.setBoolean(TAG_MANUAL_HOUSING, manualHousing);
         compound.setTag(TAG_REQUESTMANAGER, getRequestManager().serializeNBT());
         compound.setString(TAG_STYLE, style);
+        compound.setBoolean(TAG_RAIDABLE, barbarianManager.canHaveBarbEvents());
+        compound.setBoolean(TAG_AUTO_DELETE, canColonyBeAutoDeleted);
+
+        this.colonyTag = compound;
```

Comment:
Is saving this in memory and updating it every mark dirty better than just calculating it when we want it?

At the very least, if we want it stored in memory, we should remove the following from markDirty `this.writeToNBT(new NBTTagCompound());` and do something like this instead:
```
public NBTTagCompound getColonyTag()
{
    // if dirty (this is if the colonyTag is dirty, which may be different than the colony being dirty, but probably not), write to colonyTag and set not dirty then return, otherwise just return.
    return this.colonyTag;
}
```

Reply:
Had the same idea yesterday. But it doesn't change a lot, colonymanager.markDirty is called anyway which will cause a write shortly after.
But I can do that.